src/Entity/Product.php

```diff
@@ -752,42 +760,50 @@ public function addCategory(ProductProductCategory $ProductProductCategory): sel
 
     public function removeCategory(ProductProductCategory $ProductProductCategory): self
     {
-        if ($this->categories->contains($ProductProductCategory)) {
+        if ($this->categories !== null && $this->categories->contains($ProductProductCategory)) {
             $this->categories->removeElement($ProductProductCategory);
         }
 
         return $this;
     }
 
-    public function getRelatedProducts(): ProductRelatedProductCollection
+    public function getRelatedProducts(): ?ProductRelatedProductCollection
     {
         return $this->relatedProducts;
     }
 
-    public function setRelatedProducts(ProductRelatedProductCollection $collection): self
+    public function setRelatedProducts(?ProductRelatedProductCollection $collection): self
     {
-        $this->relatedProducts = new ProductRelatedProductCollection;
+        if ($collection === null) {
+            $this->relatedProducts = null;
+        } else {
```

Comment:
Iets netter zou misschien zijn om na het setten van relatedProducts naar null een `return $this` te gebruiken zodat je de else overbodig wordt? (idem voor vergelijkbare stukken code in deze PR)